lib/puppet/application/cert.rb

```diff
@@ -79,6 +79,22 @@ def subcommand=(name)
     set_log_level
   end
 
+  option("--human-readable", "-h") do |arg|
```

Comment:
I was worried that `puppet cert list -h` would show help instead of human readable extensions, but the way you describe it working sounds 👍 to me.

I've been looking into the issues with help for an entirely different reason, and in the process submitted some PRs. The `invalid option` thing will be fixed by https://github.com/puppetlabs/puppet/pull/5044. I also have some error handling fixes in https://github.com/puppetlabs/puppet/pull/5045 and https://github.com/puppetlabs/puppet/pull/5046.